### DISCUSSION TITLE
Dashboard: Compliance order

### DISCUSSION THREAD

**Discussion by K8Sudo** [monarc-project/MonarcAppFO#490] [Ideas]
With version 2.12.6, the sorting of controls in the Compliance dashboard view has changed for us. Previously, the controls were sorted clockwise. (see in the figure)

**Old:**
<img width="652" alt="Bildschirmfoto 2023-06-19 um 17 04 43" src="https://github.com/monarc-project/MonarcAppFO/assets/47662401/cf21a576-77c9-4f95-9578-49a44e52881e">

**New:**
![image](https://github.com/monarc-project/MonarcAppFO/assets/47662401/2149e396-93cf-4ffc-adfd-1a37378c8581)

Would it be possible to sort here again alphanumerically according to the "Code" field.

**Comment by ruslanbaidan** [comment]
I performed some tests on our side on the same version 2.12.6 and it seems the order is kept as before, by code.

The other observation led to a conclusion that the order by code was not applied before, however we need to validate the front-end and back-end logic more to be sure.

It could also be due to the language (DE). Would it be possible to check for you if there is an impact on english based analysis ?

**Comment by ruslanbaidan** [comment]
After deeper investigation, I can confirm that it was never ordered by code, but by ID of category of statement of applicability.
As you could see [here](https://github.com/monarc-project/ng-client/blob/master/src/ClientDashboardCtrl.js#L1921-L1936).
The order looked like by code because the created categories order was respectfully linked to the codes number.
When it comes to any imported data or manually manipulated, the order by code of measure may be broken.

**Comment by K8Sudo** [comment]
Are there already considerations, for an adjustment of the sorting in one of the next patches or is the behavior intended?

**Reply by ruslanbaidan** [reply]
There was no order by code before, but by ID of category, so the reason of the difference is related to the data import.
We are not planning to perform any release soon, but will consider to implement an order by code in the future.

**Comment by gbaumann0166** [comment]
Hello,
for us, sorting would also be advantageous. see screenshot

![image](https://github.com/monarc-project/MonarcAppFO/assets/56740147/ff320a4d-96e3-40ed-866b-a3fce3759028)
